integration/k8/geopm-prometheus.Dockerfile

```diff
@@ -1,19 +1,59 @@
-FROM ubuntu:22.04 AS env
+# Stage 1: Build container
+FROM ubuntu:24.04 AS build
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get install -yq --no-install-recommends \
+    software-properties-common python3-pip git build-essential libcap-dev \
+    libnvidia-ml-dev libgrpc-dev libgrpc++-dev libprotobuf-dev libprotoc-dev \
+    libsystemd-dev liburing-dev libtool pkgconf protobuf-compiler \
+    protobuf-compiler-grpc unzip zlib1g-dev python3-all python3-setuptools \
+    python3-setuptools-scm wget debhelper-compat dh-python curl zstd \
+    python3-cffi libze1 libze-dev python3-dev python3-defusedxml \
+    python3-build fakeroot && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy the current repository context into the container
+COPY ../../../geopm src/geopm
+
+# Build GEOPM packages
+WORKDIR /src/geopm/libgeopmd
+RUN ./autogen.sh && ./configure && ENABLE_LEVELZERO=TRUE make deb
+RUN apt-get install -yq --no-install-recommends ./*.deb
+
+WORKDIR /src/geopm/geopmdpy
+RUN ./make_deb.sh
+
+WORKDIR /src/geopm/geopmdrs
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y && \
+    export PATH="/root/.cargo/bin:${PATH}" && \
+    rustup update stable && \
+    cargo install cargo-deb && \
+    ./build.sh
+
+# Collect all .deb packages
+RUN mkdir -p /mnt/geopm-prometheus && \
+    cp -p $(find /src/geopm -name \*.deb) /mnt/geopm-prometheus
+
+# Stage 2: Runtime container
+FROM ubuntu:24.04 AS runtime
 
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update
-RUN apt-get install -yq gpg wget
-RUN wget -qO - https://repositories.intel.com/gpu/intel-graphics.key | gpg --yes --dearmor --output /usr/share/keyrings/intel-graphics.gpg
-RUN echo "deb [arch=amd64,i386 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/gpu/ubuntu jammy client" | \
-  tee /etc/apt/sources.list.d/intel-gpu-jammy.list
-RUN apt-get update
-RUN apt-get install -yq software-properties-common python3-pip
-RUN python3 -m pip install --ignore-installed grpcio==1.47.5
-COPY geopm-prometheus /mnt/geopm-prometheus
-RUN apt-get install -yq /mnt/geopm-prometheus/*.deb
+
+# Copy .deb packages from the build stage and install them
+COPY --from=build /mnt/geopm-prometheus /mnt/geopm-prometheus
+RUN apt-get install -yq --no-install-recommends /mnt/geopm-prometheus/*.deb python3-grpcio && \
+    rm -rf /mnt/geopm-prometheus
+RUN apt-get clean && \
```

Comment:
The latest change results in one image layer being added to the image that has the apt-get update result, then a second layer that has the install result, and then a third layer that removes the cache from the first two layers.  The update, install, and clean should be all part of the same RUN command.

Reply:
For what it's worth, you can also use `dpkg -i` to install individual .deb files.